node/core/dispute-coordinator/src/initialized.rs

```diff
@@ -283,37 +306,61 @@ impl Initialized {
 		self.participation.process_active_leaves_update(ctx, &update).await?;
 
 		if let Some(new_leaf) = update.activated {
-			match self
-				.rolling_session_window
-				.cache_session_info_for_head(ctx.sender(), new_leaf.hash)
-				.await
-			{
-				Err(e) => {
-					gum::warn!(
+			let session_idx =
+				self.runtime_info.get_session_index_for_child(ctx.sender(), new_leaf.hash).await;
+
+			match session_idx {
+				Ok(session_idx)
+					if self.last_consecutive_cached_session.is_none() ||
+						session_idx >
+							self.last_consecutive_cached_session.expect(
+								"The first clause explicitly handles `None` case. qed.",
+							) =>
```

Comment:
```suggestion
			let should_cache_session = |session_idx: &SessionIndex| {
				self.last_consecutive_cached_session.is_none() ||
					session_idx >
						&self
							.last_consecutive_cached_session
							.expect("The first clause explicitly handles `None` case. qed.")
			};

			match session_idx {
				Ok(session_idx) if should_cache_session(&session_idx) => {
```
Is it more readable to extract the check as a closure?

Reply:
IMO, no :)